src/hypothesis/strategies.py

```diff
@@ -412,13 +413,17 @@ def sampled_from(elements):
 
     from hypothesis.searchstrategy.misc import SampledFromStrategy, \
         JustStrategy
-    elements = tuple(iter(elements))
+    from hypothesis.internal.conjecture.utils import check_sample
     if not elements:
         return nothing()
     if len(elements) == 1:
-        return JustStrategy(elements[0])
-    else:
-        return SampledFromStrategy(elements)
+        return JustStrategy(next(iter(elements)))
```

Comment:
I've actually written such an object recently (it was a random sampler type). It's not a totally unreasonable thing to do.